runtime/src/test_utils.rs

```diff
@@ -528,7 +538,27 @@ impl<BS: Blockstore> MockRuntime<BS> {
     ) -> Result<RawBytes, ActorError> {
         self.in_call = true;
         let prev_state = self.state;
-        let res = A::invoke_method(self, method_num, params);
+        let res: Result<RawBytes, ActorError> =
+            std::panic::catch_unwind(std::panic::AssertUnwindSafe(|| {
+                A::invoke_method(self, method_num, params)
+            }))
+            .unwrap_or_else(|panic| {
+                if self.actor_exit.borrow().is_some() {
```

Comment:
Let's instead embed the exit inside the error, and cast with https://doc.rust-lang.org/std/any/trait.Any.html#method.downcast. That way, we can't accidentally swallow up a _real_ panic.

Reply:
Basically, we can use panic to "throw" the exit value.

Reply:
Does this even make sense here?

We set thr exit and immediately panic, which lands here; any other panic will be resumed. So the mentioned condition is not really possible, and it is nicer to the eye in tests since cargo will still report these panics at runtime in the console.

Now, if this was prod code, i would definitely consider the fancy way.
Here, i dont think it adds anything but complexity, really.

Having said all that, if you feel overly strongly about that and insist, i will abide.

Reply:
Eh, you're right. There's little benefit.